### PR TITLE
Add additional nested ambiguity note to GetTypeByMetadataName

### PR DESCRIPTION
**Pull request by 333fred**

### CODE REVIEW

**Comment by 333fred** [comment]
@AlekseyTs @dotnet/roslyn-compiler for review.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1). Thank you for following up!

File: src/Compilers/Core/Portable/Compilation/Compilation.cs [hunk 1]
```diff
@@ -1063,6 +1063,10 @@ public INamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
         /// a match is found in one module in an assembly, no further modules within that assembly are searched.
         /// </para>
         /// <para>Type forwarders are ignored, and not considered part of the assembly where the TypeForwardAttribute is written.</para>
+        /// <param>
```
Comment (Youssef1313) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs:1066]:
```suggestion
        /// <para>
```

File: src/Compilers/Core/Portable/Compilation/Compilation.cs [hunk 2]
```diff
@@ -1063,6 +1063,10 @@ public INamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
         /// a match is found in one module in an assembly, no further modules within that assembly are searched.
         /// </para>
         /// <para>Type forwarders are ignored, and not considered part of the assembly where the TypeForwardAttribute is written.</para>
+        /// <para>
+        /// Ambiguities are detected on each nested level. For example, if <c>A+B</c> is request, and there are multiple <c>A</c>s but only one of them has a <c>B</c> nested
```
Comment (333fred) [inline src/Compilers/Core/Portable/Compilation/Compilation.cs:1067]:
```suggestion
        /// Ambiguities are detected on each nested level. For example, if <c>A+B</c> is requested, and there are multiple <c>A</c>s but only one of them has a <c>B</c> nested
```

**Review by RikkiGibson — APPROVED** [review]

**Pull request merged**